src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,343 @@
+    this._lastValueValid = !value || this._dateAdapter.isValid(value);
+    value = this._getValidDateOrNull(value);
+    const oldDate = this.value;
+    this._value = value;
+    this._formatValue(value);
+
+    if (!this._dateAdapter.sameDate(oldDate, value)) {
+      this._valueChange.emit(value);
+    }
+  }
+  private _value: D | null;
+
+  /** The minimum valid date. */
+  @Input()
+  get min(): D | null { return this._min; }
+  set min(value: D | null) {
+    this._min = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _min: D | null;
+
+  /** The maximum valid date. */
+  @Input()
+  get max(): D | null { return this._max; }
+  set max(value: D | null) {
+    this._max = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _max: D | null;
+
+  /** Whether the datepicker-input is disabled. */
+  @Input()
+  get disabled(): boolean { return !!this._disabled; }
+  set disabled(value: boolean) {
+    const newValue = coerceBooleanProperty(value);
+    const element = this._elementRef.nativeElement;
+
+    if (this._disabled !== newValue) {
+      this._disabled = newValue;
+      this._disabledChange.emit(newValue);
+    }
+
+    // We need to null check the `blur` method, because it's undefined during SSR.
+    if (newValue && element.blur) {
+      // Normally, native input elements automatically blur if they turn disabled. This behavior
+      // is problematic, because it would mean that it triggers another change detection cycle,
+      // which then causes a changed after checked error if the input element was focused before.
+      element.blur();
+    }
+  }
+  private _disabled: boolean;
+
+  /** Emits when a `change` event is fired on this `<input>`. */
+  @Output() readonly dateChange: EventEmitter<DatepickerInputEvent<D>>;
+
+  /** Emits when an `input` event is fired on this `<input>`. */
+  @Output() readonly dateInput: EventEmitter<DatepickerInputEvent<D>>;
+
+  /** Emits when the value changes (either due to user input or programmatic change). */
+  _valueChange = new EventEmitter<D | null>();
+
+  /** Emits when the disabled state has changed */
+  _disabledChange = new EventEmitter<boolean>();
+
+  /** Implemented as part of ControlValueAccessor. */
+  _onTouched = () => {};
+
+  /** Implemented as part of ControlValueAccessor. */
+  private _controlValueAccessorOnChange: (value: any) => void = () => {};
+
+  /** Implemented as part of ControlValueAccessor. */
+  private _validatorOnChange = () => {};
+
+  /** Implemented for datepicker CDK and locale subscriptions. */
+  private readonly _subscriptions = new Subscription();
+
+  /** The form control validator for whether the input parses. */
+  private _parseValidator: ValidatorFn = (): ValidationErrors | null => {
+    return this._lastValueValid ?
+        null : {cdkDatepickerParse: {text: this._elementRef.nativeElement.value}};
```

Comment:
I think we actually do want the name of the error (e.g. 'cdkDatepickerParse') quoted to prevent closure from changing it. This is the same thing Angular Forms does with their built in error names

Reply:
Will quote the name of the error (e.g. `cdkDatepickerParse`), but will leave the other property (e.g. 'text') unquoted.

Reply:
Looking at how `@angular/forms` does it, I think we should quote both: https://github.com/angular/angular/blob/master/packages/forms/src/validators.ts#L233